CLI: Add the command `verdi presto` (#6351)
aiidateam/aiida-core: docs/source/reference/command_line.rst

Reviewer: This is a reference documentation. Should this new command also be mentioned in tutorials and other places throughout? (perhaps in a separate PR so we can start dogfooding this).

Author: Yes, this should definitely be done, but I was leaving that for a separate PR. This would require significant changes in the documentation structure so want to tackle that separately

      list  Display a list of all available plugins.

.. _reference:command-line:verdi-presto:

``verdi presto``
----------------

.. code:: console

    Usage:  [OPTIONS]

>       Set up a new profile as automatically as possible.

Reviewer: This sentence sounds strange to me. Maybe...

```suggestion
      Set up a new profile without a fuss.
```

Or 
```suggestion
      Set up a new, pre-configured profile.
```

```suggestion
    Quickly set up a new profile.
```

Not quite happy with any of these, perhaps @mbercx will have the right idea.

Author: Also found it difficult to find a good phrasing for this. Agree that what I have currently is unsatisfactory

Reviewer: Given the name of the command, perhaps something a bit light-hearted might actually work. :-)

Author: >  Given the name of the command, perhaps something a bit light-hearted might actually work. :-)

Yeah fair enough, how about

> Set up a new profile in a jiffy.

Not sure how well known that expression is to non-Brits though

Reviewer: I _love_ `in a jiffy`! But as you mention, it probably might confuse some people. And given `presto` itself might not be known to everybody (sorry italians :it: ) we better be clear in the first sentence of the docstring.

I asked [a hippo](https://www.wordhippo.com/what-is/another-word-for/in_a_jiffy.html) for help, how about

```
Set up a profile in no time.
Set up a profile without hassle.
Set up a profile rapidly/swiftly and without hassle.
```

The last one tries to convey that it should be both quick and easy (without a fuss). But fuss or hassle might have the same problem as jiffy? Okay I'll stop painting the bikeshed now. :bike:

Reviewer 2: It's an excellent question which clearly requires extensive debate. ^^

I think one issue in properly describing the command is that it doesn't set up a barebones profile because it _does_ check for a RabbitMQ broker, but it doesn't check for all services since for PostgreSQL that is more complicated.

However, the most important aspect of the `verdi presto` command is that it's fast. So I'm in favor of something like:

```
Set up a new profile in seconds.
```

The second aspect is that it doesn't require user input (for now). So we could add "automatically":

```
Automatically set up a new profile in seconds.
```

I'm also fine with "in no time", "without hassle", "in a jiffy". But I think "in seconds" is the easiest to understand for non-natives?

Reviewer 2: We could also use "in a matter of seconds" or "in a few seconds". But maybe we should block a 3 hour Zoom call to discuss this at length... 😁

>       This command aims making setting up a new profile as easy as possible. It intentionally
>       provides a limited amount of options to customize the profile and by default does not
>       require any options to be specified at all. For full control, please use the command
>       `verdi profile setup` instead.

Reviewer: ```suggestion
      This command aims to make setting up a new profile as easy as possible. It intentionally
      provides only a limited amount of options to customize the profile, and by default does not
      require any options to be specified at all. For full control, please use 
      `verdi profile setup`.
```

>       The main goal for this command is to setup a profile with minimal requirements to make
>       it easy to install AiiDA and get started as quickly as possible. Therefore, by default,
>       the created profile uses the `core.sqlite_dos` storage plugin which does not require any
>       services, such as PostgreSQL and RabbitMQ are not required. This _does_ mean, however,
>       that not all functionality of AiiDA is available, most notably running the daemon and
>       submitting processes to said daemon.

Reviewer: There is some repetition with respect with the previous paragraph, suggested rewording.

```suggestion
      After running `verdi presto` you can immediately start using AiiDA without additional setup. 
      The created profile uses the `core.sqlite_dos` storage plugin which does not require any
      services, such as PostgreSQL and RabbitMQ. This _does_ mean, however,
      that not all functionality of AiiDA is available, most notably running the daemon and
      submitting long-running processes to said daemon.
```

Author: I removed "long-running" processes from your suggestion. I get the point, but it suggests that some other processes might be submitted, but there is no submission whatsoever.

Reviewer: Makes sense. Although I just realized that the given text is still not quite accurate, since we're mixing the DB plugin (which is just a question of performance) with the presence of rabbitmq, which is what actually determines whether we can have a darmon or not! And now you're even detecting whether rabbitmq is present, and if it is, all is actually fine. So I think a bit more nuance here is needed.

Author: The plan is to extend this command with the functionality of `verdi quicksetup` to automatically create resources needed for the `core.psql_dos`. So the comment will become more incorrect. But I will already update it now for the fact that it tries to configure RabbitMQ if it can connect to it. BEcause that is already in

      The command performs the following actions:

      * Create a new profile that is set as the new default

>       * Create a default user for the profile (email can be configured through options)

Reviewer: ```suggestion
      * Create a default user for the profile (email can be configured through the `--email` option)
```

      * Set up the localhost as a `Computer` and configure it

>       * Set a number of configuration options with sensible defaults

Reviewer: I wonder if there is a small set of most impactful configuration options that should be mentioned? Or at least we could link to some section where more info can be found (maybe the `verdi profile setup` command)?

Author: You mean just list here the options that are set? And why the `verdi profile setup` command? That doesn't allow directly to set configuration options.

Reviewer: Yeah, just mention the options that are set. But feel free to ignore this.

Tbh I am still really confused about the difference between profile setup and configuration. Why is there a split like that? (Convo for another time, please ignore)

    Options:
      --profile-name TEXT  The name of the profile. By default generates a name that does not
                           already exist and starts with `presto`.  [default: (dynamic)]

>       --email TEXT         The email of the default user.  [default: aiida@localhost]

Reviewer: ```suggestion
      --email TEXT         Email of the default user.  [default: aiida@localhost]
```
